includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -485,11 +490,50 @@ public function process_payment_for_order( $args ) {
 
 			$intent_id = $intent->get_id();
 			$status    = $intent->get_status();
+			$charge_id = $intent->get_charge_id();
+		} else {
+			// For $0 orders, we need to save the payment method using a setup intent.
+			$intent = $this->payments_api_client->create_setup_intent(
+				$payment_method,
+				$customer_id,
+				'true'
+			);
+
+			$intent_id = $intent['id'];
+			$status    = $intent['status'];
+			$charge_id = '';
 
+			// In SCA cases the setup intent status might be requires_action and we should display the authentication modal.
+			// For now, since we're not supporting SCA cards, we can ignore that status.
+			if ( 'succeeded' !== $status ) {
+				throw new Exception( __( 'Failed to add the provided payment method. Please try again later', 'woocommerce-payments' ) );
+			}
+		}
+
+		if ( ! empty( $intent ) ) {
 			if ( 'succeeded' !== $status && 'requires_capture' !== $status ) {
 				$intent_failed = true;
 			}
 
+			if ( $save_payment_method && ! $intent_failed ) {
+				try {
+					$token = $this->token_service->add_payment_method_to_user( $payment_method, $user );
+				} catch ( Exception $e ) {
+					// If saving the token fails, log the error message but catch the error to avoid crashing the checkout flow.
+					Logger::log( 'Error when saving payment method: ' . $e->getMessage() );
```

Comment:
If I am looking at the logs, will it be clear that this error originated here and not from update_order_status (which uses the exact same error)

Reply:
This change is just some re-ordering of the saved cards code, but I'm not sure if they should have different wordings, as both are related to saving the payment method. The code in `update_order_status` runs for SCA scenarios but does the same thing as the one in `process_payment_for_order`.

Would you prefer that those messages are changed to reflect if the card is an SCA or a basic one? If so, I think it would be better to do this in another PR, as the change is not really related to this PR's scope.